atom/browser/native_window_mac.mm

```diff
@@ -95,6 +95,33 @@ - (void)windowDidResignMain:(NSNotification*)notification {
   shell_->NotifyWindowBlur();
 }
 
+- (NSSize)windowWillResize:(NSWindow *)sender toSize:(NSSize)frameSize {
+  NSSize newSize = frameSize;
+  double interiorContentAspectRatio = shell_->GetInteriorContentAspectRatio();
+
+  if (interiorContentAspectRatio > 0.0) {
+    NSRect windowFrame = [sender frame];
+    NSRect contentViewBounds = [[sender contentView] bounds];
```

Comment:
The contentView of Electron's windows is a [FullSizeContentView](https://github.com/atom/electron/blob/master/atom/browser/native_window_mac.mm#L29-L52) which always resize itself to the size of superView, resulting [the size of contentView including window frame](https://github.com/atom/electron/issues/1024).

> Is all of the frameless window implementation within native_window_mac.mm?

Yeah.